### ISSUE TITLE
bug: failed fuzzing test

### ISSUE THREAD

**Issue by zwang28** [risingwavelabs/risingwave#12377]
### Describe the bug

kernel bug or unsupported SQL?

https://buildkite.com/risingwavelabs/pull-request/builds/31832#018aa739-f3b2-47cc-abbd-f226df7c2d21

[fuzzing-22.txt](https://github.com/risingwavelabs/risingwave/files/12645651/fuzzing-22.txt)

```
2022-03-27T06:23:35.029182Z  INFO node{id=14 name="client"}:task{id=581}: risingwave_sqlsmith::runner: [EXECUTING STATEMENT]: SELECT (BIGINT '629') AS col_0, t_1.n_comment AS col_1 FROM m3 AS t_0 JOIN nation AS t_1 ON t_0.col_1 = t_1.n_comment WHERE EXISTS (WITH with_2 AS (SELECT t_4.o_comment AS col_0 FROM hop(alltypes1, alltypes1.c11, INTERVAL '33', INTERVAL '165') AS hop_3 FULL JOIN orders AS t_4 ON hop_3.c3 = t_4.o_shippriority AND true AND hop_3.c7 = t_4.o_totalprice, (WITH with_6 AS (SELECT hop_8.c15 AS col_0, t_12.c5 AS col_1, t_7.p_brand AS col_2 FROM part AS t_7 LEFT JOIN hop(alltypes2, alltypes2.c11, INTERVAL '54', INTERVAL '648') AS hop_8 ON t_7.p_partkey = hop_8.c3 AND true AND t_7.p_mfgr = hop_8.c9 AND t_7.p_retailprice = hop_8.c7, ((SELECT t_9.col_3 AS col_0, (coalesce(NULL, NULL, NULL, NULL, NULL, t_9.col_1, NULL, NULL, NULL, NULL)) AS col_1, true AS col_2 FROM m11 AS t_9 GROUP BY t_9.col_3, t_9.col_1) AS sq_10 JOIN m19 AS t_11 ON sq_10.col_0 = t_11.col_2 AND ((SMALLINT '837') <= (INT '649'))) LEFT JOIN alltypes1 AS t_12 ON sq_10.col_1 = t_12.c11 AND true GROUP BY GROUPING SETS ((t_12.c3, t_12.c13, t_12.c8, t_7.p_container, t_12.c6, t_12.c5, t_11.col_0, hop_8.c15, t_12.c4, hop_8.c4), (t_12.c16, t_12.c3, t_12.c8, t_12.c6, hop_8.c10, t_11.col_2, hop_8.c4, t_7.p_retailprice, t_12.c2), (t_12.c7, hop_8.c6, t_12.c15, t_12.c9, t_11.col_1, sq_10.col_1, t_12.c6, t_12.c1, hop_8.c5, t_12.c14, hop_8.c15, t_11.col_2, hop_8.c4, hop_8.c10, t_12.c11, t_7.p_partkey, hop_8.c1, t_7.p_brand, t_7.p_retailprice)) HAVING (CASE WHEN t_12.c1 THEN t_12.c1 WHEN hop_8.c1 THEN hop_8.c1 ELSE t_12.c1 END) LIMIT 40) SELECT (FLOAT '957') AS col_0, ('0bFQK2lbzN') AS col_1 FROM with_6 WHERE false LIMIT 54) AS sq_13 WHERE hop_3.c1 GROUP BY t_4.o_orderpriority, hop_3.c16, hop_3.c2, hop_3.c10, t_4.o_totalprice, t_4.o_comment, hop_3.c6, hop_3.c14, hop_3.c4, sq_13.col_1, hop_3.c8, hop_3.c13, hop_3.c1 HAVING hop_3.c1 LIMIT 29) SELECT t_1.n_name AS col_0, (FLOAT '418') AS col_1 FROM with_2 GROUP BY t_1.n_name, t_0.col_2, t_0.col_0, t_1.n_regionkey ORDER BY t_1.n_name DESC, t_0.col_0 DESC, t_0.col_2 DESC, t_1.n_name ASC LIMIT 86) GROUP BY t_1.n_comment
thread '<unnamed>' panicked at 'internal error: entered unreachable code: CorrelatedInputRef CorrelatedInputRef { index: 15, correlated_id: 0 } has not been decorrelated', src/frontend/src/expr/correlated_input_ref.rs:84:9
stack backtrace:
   0: rust_begin_unwind
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/std/src/panicking.rs:593:5
   1: core::panicking::panic_fmt
             at /rustc/f0411ffcebcd7f75ac02ed45feb53ffd07b75398/library/core/src/panicking.rs:67:14
   2: <risingwave_frontend::expr::correlated_input_ref::CorrelatedInputRef as risingwave_frontend::expr::Expr>::to_expr_proto
             at ./src/frontend/src/expr/correlated_input_ref.rs:84:9
   3: <risingwave_frontend::expr::ExprImpl as risingwave_frontend::expr::Expr>::to_expr_proto
             at ./src/frontend/src/expr/mod.rs:109:45
   4: <risingwave_frontend::optimizer::plan_node::stream_hop_window::StreamHopWindow as risingwave_frontend::optimizer::plan_node::to_prost::StreamNode>::to_stream_prost_body::{{closure}}
             at ./src/frontend/src/optimizer/plan_node/stream_hop_window.rs:120:26
```

**Comment by st1page** [comment]
can not reproduce it in the latest main branch. Maybe it has been fixed?

**Comment by st1page** [comment]
reproduce:
```SQL
CREATE TABLE alltypes1 (
    c1 boolean,
    c2 smallint,
    c3 integer,
    c4 bigint,
    c5 real,
    c6 double precision,
    c7 numeric,
    c8 date,
    c9 varchar,
    c10 time without time zone,
    c11 timestamp without time zone,
    -- TODO(Noel): Many timestamptz expressions are unsupported in RW, leave this out for now.
    -- c12 timestamp with time zone,
    c13 interval,
    c14 struct < a integer >,
    c15 integer [],
    c16 varchar []
);

CREATE TABLE alltypes2 (
    c1 boolean,
    c2 smallint,
    c3 integer,
    c4 bigint,
    c5 real,
    c6 double precision,
    c7 numeric,
    c8 date,
    c9 varchar,
    c10 time without time zone,
    c11 timestamp without time zone,
    -- TODO(Noel): Many timestamptz expressions are unsupported in RW, leave this out for now.
    -- c12 timestamp with time zone,
    c13 interval,
    c14 struct < a integer >,
    c15 integer [],
    c16 varchar []
);

CREATE TABLE bid (
    auction BIGINT,
    bidder BIGINT,
    price BIGINT,
    channel VARCHAR,
    url VARCHAR,
    date_time TIMESTAMP,
    extra VARCHAR
);

CREATE TABLE nation (
    n_nationkey INTEGER,
    n_name VARCHAR,
    n_regionkey INTEGER,
    n_comment VARCHAR,
    PRIMARY KEY (n_nationkey)
);

CREATE TABLE orders (
    o_orderkey BIGINT,
    o_custkey INTEGER,
    o_orderstatus VARCHAR,
    o_totalprice NUMERIC,
    o_orderdate DATE,
    o_orderpriority VARCHAR,
    o_clerk VARCHAR,
    o_shippriority INTEGER,
    o_comment VARCHAR,
    PRIMARY KEY (o_orderkey)
);

CREATE TABLE part (
    p_partkey INTEGER,
    p_name VARCHAR,
    p_mfgr VARCHAR,
    p_brand VARCHAR,
    p_type VARCHAR,
    p_size INTEGER,
    p_container VARCHAR,
    p_retailprice NUMERIC,
    p_comment VARCHAR,
    PRIMARY KEY (p_partkey)
);

CREATE TABLE person (
    id BIGINT,
    name VARCHAR,
    email_address VARCHAR,
    credit_card VARCHAR,
    city VARCHAR,
    state VARCHAR,
    date_time TIMESTAMP,
    extra VARCHAR,
    PRIMARY KEY (id)
);

CREATE MATERIALIZED VIEW m11 AS SELECT (((SMALLINT '20') | ((BIGINT '9223372036854775807') / (INT '1428609145'))) - (BIGINT '1')) AS col_0, hop_4.date_time AS col_1, hop_4.date_time AS col_2, hop_4.date_time AS col_3 FROM (SELECT 'sU9fJxX40J' AS col_0, t_2.name AS col_1 FROM (tumble(alltypes1, alltypes1.c11, INTERVAL '8') AS tumble_0 FULL JOIN hop(person, person.date_time, INTERVAL '6', INTERVAL '102') AS hop_1 ON tumble_0.c9 = hop_1.state AND true) RIGHT JOIN person AS t_2 ON hop_1.city = t_2.name AND true AND hop_1.extra = t_2.email_address AND hop_1.credit_card = t_2.name AND hop_1.city = t_2.state AND hop_1.city = t_2.credit_card AND tumble_0.c11 = t_2.date_time AND hop_1.extra = t_2.city AND hop_1.city = t_2.email_address AND tumble_0.c9 = t_2.state AND hop_1.state = t_2.state AND tumble_0.c9 = t_2.email_address AND hop_1.name = t_2.state AND hop_1.credit_card = t_2.city WHERE tumble_0.c1 GROUP BY t_2.name, hop_1.credit_card) AS sq_3 LEFT JOIN hop(bid, bid.date_time, INTERVAL '32', INTERVAL '64') AS hop_4 ON sq_3.col_1 = hop_4.url AND true GROUP BY hop_4.date_time HAVING ((INT '948') <> (BIGINT '291'));
CREATE MATERIALIZED VIEW m19 AS SELECT tumble_2.auction AS col_0, tumble_2.channel AS col_1, tumble_2.date_time AS col_2 FROM tumble(bid, bid.date_time, INTERVAL '9') AS tumble_2 GROUP BY tumble_2.channel, tumble_2.url, tumble_2.date_time, tumble_2.auction HAVING true
SELECT (BIGINT '629') AS col_0, t_1.n_comment AS col_1 FROM m3 AS t_0 JOIN nation AS t_1 ON t_0.col_1 = t_1.n_comment WHERE EXISTS (WITH with_2 AS (SELECT t_4.o_comment AS col_0 FROM hop(alltypes1, alltypes1.c11, INTERVAL '33', INTERVAL '165') AS hop_3 FULL JOIN orders AS t_4 ON hop_3.c3 = t_4.o_shippriority AND true AND hop_3.c7 = t_4.o_totalprice, (WITH with_6 AS (SELECT hop_8.c15 AS col_0, t_12.c5 AS col_1, t_7.p_brand AS col_2 FROM part AS t_7 LEFT JOIN hop(alltypes2, alltypes2.c11, INTERVAL '54', INTERVAL '648') AS hop_8 ON t_7.p_partkey = hop_8.c3 AND true AND t_7.p_mfgr = hop_8.c9 AND t_7.p_retailprice = hop_8.c7, ((SELECT t_9.col_3 AS col_0, (coalesce(NULL, NULL, NULL, NULL, NULL, t_9.col_1, NULL, NULL, NULL, NULL)) AS col_1, true AS col_2 FROM m11 AS t_9 GROUP BY t_9.col_3, t_9.col_1) AS sq_10 JOIN m19 AS t_11 ON sq_10.col_0 = t_11.col_2 AND ((SMALLINT '837') <= (INT '649'))) LEFT JOIN alltypes1 AS t_12 ON sq_10.col_1 = t_12.c11 AND true GROUP BY GROUPING SETS ((t_12.c3, t_12.c13, t_12.c8, t_7.p_container, t_12.c6, t_12.c5, t_11.col_0, hop_8.c15, t_12.c4, hop_8.c4), (t_12.c16, t_12.c3, t_12.c8, t_12.c6, hop_8.c10, t_11.col_2, hop_8.c4, t_7.p_retailprice, t_12.c2), (t_12.c7, hop_8.c6, t_12.c15, t_12.c9, t_11.col_1, sq_10.col_1, t_12.c6, t_12.c1, hop_8.c5, t_12.c14, hop_8.c15, t_11.col_2, hop_8.c4, hop_8.c10, t_12.c11, t_7.p_partkey, hop_8.c1, t_7.p_brand, t_7.p_retailprice)) HAVING (CASE WHEN t_12.c1 THEN t_12.c1 WHEN hop_8.c1 THEN hop_8.c1 ELSE t_12.c1 END) LIMIT 40) SELECT (FLOAT '957') AS col_0, ('0bFQK2lbzN') AS col_1 FROM with_6 WHERE false LIMIT 54) AS sq_13 WHERE hop_3.c1 GROUP BY t_4.o_orderpriority, hop_3.c16, hop_3.c2, hop_3.c10, t_4.o_totalprice, t_4.o_comment, hop_3.c6, hop_3.c14, hop_3.c4, sq_13.col_1, hop_3.c8, hop_3.c13, hop_3.c1 HAVING hop_3.c1 LIMIT 29) SELECT t_1.n_name AS col_0, (FLOAT '418') AS col_1 FROM with_2 GROUP BY t_1.n_name, t_0.col_2, t_0.col_0, t_1.n_regionkey ORDER BY t_1.n_name DESC, t_0.col_0 DESC, t_0.col_2 DESC, t_1.n_name ASC LIMIT 86) GROUP BY t_1.n_comment

```

**Comment by st1page** [comment]
fixed by #12341
